### PR TITLE
update gitpython to 3.1.41 (ltm)

### PR DESCRIPTION
**Pull request by gsnider2195**
Backport of #5083

### CODE REVIEW

File: poetry.lock [hunk 1]
```diff
@@ -2988,54 +2988,54 @@ files = [
 
 [[package]]
 name = "pyuwsgi"
-version = "2.0.23"
+version = "2.0.23.post0"
```
Comment (glennmatthews) [inline poetry.lock:2991]: A bit hesitant about this change as there doesn't seem to be much documentation around it. I *think* this is primarily to fix a compatibility issue with Python 3.12

File: poetry.lock [hunk 2]
```diff
@@ -1633,13 +1633,13 @@ colors = ["colorama (>=0.4.6)"]
 
 [[package]]
 name = "jinja2"
-version = "3.1.2"
+version = "3.1.3"
```
Comment (glennmatthews) [inline poetry.lock:1636]: Want to also update this in pyproject.toml (#5086 / CVE-2024-22195)?

File: poetry.lock (hunk 1 above)
Comment (gsnider2195) [reply poetry.lock]: Yeah I forgot to do `--no-update` when I updated the lockfile. Fixed now

File: nautobot/docs/requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 griffe==0.31.0
-Jinja2==3.1.2
+Jinja2==3.1.3
```
Comment (glennmatthews) [inline nautobot/docs/requirements.txt:2]: does the example_plugin also need this update or is that 2.x-only?

**Review by glennmatthews — APPROVED** [review]

File: nautobot/docs/requirements.txt (hunk 3 above)
Comment (gsnider2195) [reply nautobot/docs/requirements.txt:2]: I missed it. It's updated now

**Review by glennmatthews — APPROVED** [review]

**Review by whitej6 — APPROVED** [review]

**Pull request merged**